torch/csrc/distributed/c10d/TCPStoreLibUvBackend.cpp

```diff
@@ -219,8 +219,8 @@ class UvTcpServer : public UvTcpSocket {
     res->handleReady();
     try {
       int uv_res = uv_tcp_open((uv_tcp_t*)res->unsafeGetStream(), socket);
-      TORCH_CHECK_WITH(
-          DistStoreError,
+      C10D_CHECK_WITH(
```

Comment:
For my learning purpose, when set `master_listen_fd`, we will fail this `C10D_CHECK_WITH`? why the `uv_listen` will be reached?

Reply:
Setting `master_listen_fd` goes through the `makeWithSocket` function which until now hasn't been used. 

This CHECK_WITH call will pass if it's a valid FD. This is just creating the tcp_t type from the provided FD.

The `uv_listen` below is required to bind the `on_new_connection` function to the provided socket. Without these changes we will bind the socket but never listen/accept connections from it

Reply:
Ah, makes sense.. Thanks for your detailed explanation.